Post-API.js

```diff
@@ -0,0 +1,48 @@
+const express=require("express");
+const XRegExp=require("xregexp");
+const app=express();
+app.use(express.json());
+
+referenceData=[]
+
+app.post("/post",(req,res)=>{
+
+    var reqBody=req.body;
+    referenceData=reqBody.referenceData;
+    var transform= JSON.stringify(reqBody.payload)
+    for(ref in referenceData){
+        transform=transform.replace(XRegExp(`{${ref}}`),referenceData[ref]);
```

Comment:
this code works. 
but big string variable(transform) will be loaded in to RAM, causing significant load on RAM for every refData search/ replace. if we 'parse' the json object, we check if ref is string and only then we replace so RAM load will be significantly less.
Can you change the code to replace using walk through json and replace.
 
Also, here is an advanced 'stream' based method. try if you can figure out that and implement that.

Reply:
Ok sure. I have pushed latest code to walk through json and replace. This code wont load RAM as the previous one.